apps/api/src/app/auth/framework/auth.decorator.ts

```diff
@@ -0,0 +1,15 @@
+import { ApiBearerAuth } from '@nestjs/swagger';
+import { applyDecorators, UseGuards } from '@nestjs/common';
+import { BEARER_SWAGGER_SECURITY_NAME } from '@novu/application-generic';
+import { isClerkEnabled } from '@novu/shared';
+import { CommunityUserAuthGuard } from './community.user.auth.guard';
+
+export function RequireAuthentication() {
+  if (isClerkEnabled()) {
+    const { EEUserAuthGuard, EEPermissionsGuard } = require('@novu/ee-auth');
+
+    return applyDecorators(UseGuards(EEUserAuthGuard, EEPermissionsGuard), ApiBearerAuth(BEARER_SWAGGER_SECURITY_NAME));
+  }
+
+  return applyDecorators(UseGuards(CommunityUserAuthGuard), ApiBearerAuth(BEARER_SWAGGER_SECURITY_NAME));
+}
```

Comment:
The `PermissionGuard` is always paired with `UserGuard`.

- `PermissionGuard` → requires `UserGuard` (it decodes the JWT and supplies the permissions)
- `UserGuard` → does **not** automatically require `PermissionGuard`

For authenticated routes that **don’t** need permission checks, add the `@SkipPermissionsCheck()` decorator.  
This keeps the code **concise** and follows the principle of least privilege.

Manually applying `PermissionGuard` to each route would be verbose and error-prone, easy to forget and accidentally leave an endpoint unprotected.

Ideally, both guards would be globally registered, and unauthenticated routes (e.g., webhooks) would be marked with `@PublicEndpoint`.  
But that’s a story for another day.